src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        // Convert single date strings to LocalDate objects
```

Comment:
First, remove comments from your code.
Second, let's see another way of implementation: 
```
LocalDate fromDate = convertToDate(dateFrom);
LocalDate toDate = convertToDate(dateTo);
StringBuilder report = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
for (String name : names) {
       int totalSalary = 0;
       for (String values : data) {
           String[] rowDataArray = rowData.split(" ");
           if ((separate method to validate date in range) && name.equals(rowDataArray[1]) {
                totalSalary += Integer.parseInt(rowDataArray[2]) * Integer.parseInt(rowDataArray[3]);
           }
report.append()....
return report;
```